Added Cuprite adapter

espen opened the pull request.

espen committed:
Added Cuprite adapter

# Conflicts:
#	Gemfile.lock

espen commented:
Thoughts on this @nruth ? The bot has some complains but IIRC the files are based on the existing adapters/tests and follow the same style.

dependabot[bot] committed:
Bump addressable from 2.7.0 to 2.8.0

Bumps [addressable](https://github.com/sporkmonger/addressable) from 2.7.0 to 2.8.0.
- [Release notes](https://github.com/sporkmonger/addressable/releases)
- [Changelog](https://github.com/sporkmonger/addressable/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sporkmonger/addressable/compare/addressable-2.7.0...addressable-2.8.0)

---
updated-dependencies:
- dependency-name: addressable
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

olleolleolle committed:
gemspec: remove unused directives

This gem exposes no executables and test_files is not used by RubyGems.org.

olleolleolle committed:
docs: Fix formatting

I think there was a hard space, so I removed it and added a formatting newline.

Commit:
Fix incompatibility with rack-test 2.0

https://github.com/rack/rack-test/pull/297

Commit:
Remove poltergeist support

Commit:
update Gemfile.lock dev dependencies

Commit:
dev dependency: unlock webdrivers and upgrade to v5
require false in gemfile as per its install instructions

Commit:
update semaphore badge/link urls

Commit:
fix Gemfile.lock reference to own version

olleolleolle committed:
test: Configure RSpec to exclude top-level DSL

Exclude the top-level DSL from Object and such; require the top-level
RSpec constructs to start with "RSpec.".

nruth commented:
I'm not familiar with cuprite, but first impressions are that it looks ok to add. I'll take another look. 

Have you been using this registered as a custom driver? Does that work ok?

Commit:
changelog rack-test 2 fix

olleolleolle committed:
Use SVG badge in README

SVG badges are... pretty.

espen commented:
Yes I am using it. Rebased and added webrick commit for Ruby 3. However I was unable to run all tests now due to some Firefox issue

dependabot[bot] committed:
Bump nokogiri from 1.13.7 to 1.14.3 (#76)

Bumps [nokogiri](https://github.com/sparklemotion/nokogiri) from 1.13.7 to 1.14.3.
- [Release notes](https://github.com/sparklemotion/nokogiri/releases)
- [Changelog](https://github.com/sparklemotion/nokogiri/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sparklemotion/nokogiri/compare/v1.13.7...v1.14.3)

---
updated-dependencies:
- dependency-name: nokogiri
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
Add webrick gem to dev dependencies (for testing the gem)
It's no longer stdlib in modern ruby

Commit:
drop webdrivers gem, use selenium-webdriver manager directly

Commit:
bundle update sinatra

Commit:
bundle update capybara

Commit:
bundle update selenium-webdriver

Adds a readme note on using chrome-for-testing with an env path
to work around chromedriver vs chrome version issues in running the gem
tests, which seemed to get worse with this update

Commit:
bundle update

espen committed:
add rackup
